flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/io/StreamTwoInputProcessor.java

```diff
@@ -124,6 +120,7 @@ private int selectFirstReadingInputIndex() throws IOException {
 	}
 
 	private void checkFinished(InputStatus status) throws Exception {
+		updateAvailability();
```

Comment:
why do we need this? The currently processed input should be updated via the returned `InputStatus`, no need to check it. If the other input is unavailable, and both will become unavailable, we would do the full check via waiting on availability future and we update the inputs (via cheap `isAproximatelyAvailable()`) once future is completed.

Reply:
The commit is unnecessary. I did not fully understand the availability handling.